io-engine/src/grpc/v1/replica.rs

```diff
@@ -532,10 +536,41 @@ impl ReplicaRpc for ReplicaService {
         .await
     }
 
+    #[named]
     async fn set_replica_entity_id(
         &self,
-        _request: Request<SetReplicaEntityIdRequest>,
+        request: Request<SetReplicaEntityIdRequest>,
     ) -> GrpcResult<Replica> {
-        unimplemented!()
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                info!("{args:?}");
+                let rx = rpc_submit::<_, _, LvsError>(async move {
+                    if let Some(bdev) =
+                        UntypedBdev::lookup_by_uuid_str(&args.uuid)
+                    {
+                        let lvol = Lvol::try_from(bdev)?;
+                        let _ = lvol
+                            .set_blob_attr("entity_id", args.entity_id, true)
```

Comment:
Here also need to catch the error return from `set_blob_attr` function and return back proper error code to control-plane. Possibly a new error code can be introduced.

Reply:
Should control plane retry with backoff? It can hold off volume_publish call btw...

Reply:
Do We want to success the volume publish with this grpc failure ? Then how this attribute can be credible for the replica in control-plane, as in some scenario, volume can published without setting the attribute.

Reply:
As discussed, We are currently using entity_id for stats metadata. So better we handle error silently.

Reply:
Seems a bit to not fail here when the one outcome we want to from this is to set the entity id!??